tidy3d/components/base.py

```diff
@@ -180,6 +180,20 @@ class Config:
 
     _cached_properties = pydantic.PrivateAttr({})
 
+    attrs: dict = pydantic.Field(
+        {},
+        title="Attributes",
+        description="Dictionary storing arbitrary metadata for a Tidy3D object. "
+        "This dictionary can be freely used by the user for storing data without affecting the "
+        "operation of Tidy3D as it is not used internally. "
+        "Note that, unlike regular Tidy3D fields, ``attrs`` are mutable. "
+        "For example, the following is allowed for setting an ``attr`` ``obj.attrs['foo'] = bar``. "
+        "Also note that if objects that are not serializable (eg types) are added to "
```

Comment:
what do you mean by "eg types"? Like if I have something like `[Callable]` in my attrs?

Reply:
I mean that you can't put `int` in your attrs, e.g. `attrs = {'test': int}`

Reply:
I just removed this `eg types` comment because it might just be confusing

Reply:
```
Also note that `Tidy3D`` will raise a ``TypeError`` if ``attrs`` contain objects that can not be serialized.
```